partitioning: Corrected partition management (#5498)
pingcap/docs: partitioned-table.md

## Partition management

> For `LIST` and `RANGE` partitioned tables, you can add and drop partitions with `ALTER TABLE <table name> ADD PARTITION (<partition specification>)` / `ALTER TABLE <table name> DROP PARTITION <list of partitions>`.

Reviewer: ```suggestion
For `LIST` and `RANGE` partitioned tables, you can add and drop partitions using the `ALTER TABLE <table name> ADD PARTITION (<partition specification>)` or `ALTER TABLE <table name> DROP PARTITION <list of partitions>`. statement.
```

For `HASH` partitioned tables, `COALESCE PARTITION` and `ADD PARTITION` is not yet supported.

`REORGANIZE PARTITION` is not yet supported.

> > **Warning**
> > `EXCHANGE PARTITION` is an experimental feature and should not be used in production environment, it can be enabled by setting `tidb_enable_exchange_partition` system variable to ON.
> > 
> > It works by swapping a partition and a non partitioned table, similar to how rename table like `RENAME TABLE t1 TO t1_tmp, t2 TO t1, t1_tmp TO t2` works.
> > `ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table`

Reviewer: What is `ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table`? Any description for it?
```suggestion
> **Warning:**
>
> - `EXCHANGE PARTITION` is an experimental feature. It is not recommended to use it in a production environment. To enable it, set the `tidb_enable_exchange_partition` system variable to `ON`.
> 
> - It works by swapping a partition and a non-partitioned table, similar to how renaming a  table like `RENAME TABLE t1 TO t1_tmp, t2 TO t1, t1_tmp TO t2` works.
>
> - `ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table`
```

## Partition management

For `LIST` and `RANGE` partitioned tables, you can add and drop partitions using the `ALTER TABLE <table name> ADD PARTITION (<partition specification>)` or `ALTER TABLE <table name> DROP PARTITION <list of partitions>` statement.

> For `HASH` partitioned tables, `COALESCE PARTITION` and `ADD PARTITION` is not yet supported.

Reviewer: ```suggestion
For `HASH` partitioned tables, `COALESCE PARTITION` and `ADD PARTITION` are not yet supported.
```

Author: Updated, thanks for spotting the bad grammar.

`REORGANIZE PARTITION` is not yet supported.

> **Warning:**
>
> - `EXCHANGE PARTITION` is an experimental feature. It is not recommended to use it in a production environment. To enable it, set the `tidb_enable_exchange_partition` system variable to `ON`.
> 

> > - It works by swapping a partition and a non-partitioned table, similar to how renaming a  table like `RENAME TABLE t1 TO t1_tmp, t2 TO t1, t1_tmp TO t2` works.

Reviewer: What does "it" refer to here?

Reviewer: ```suggestion
> - `EXCHANGE PARTITION` works by swapping a partition and a non-partitioned table, similar to how renaming a  table like `RENAME TABLE t1 TO t1_tmp, t2 TO t1, t1_tmp TO t2` works.
```

Author: Thanks for the suggestion, I have updated it.

>

> > - `ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table`

Reviewer: What does this statement mean?

Author: Changed it to an example:
```As an example: `ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table`, swaps the table `non_partitioned_table` with the partition `p1` in the `partitioned_table` table.```

## Partition management

For `LIST` and `RANGE` partitioned tables, you can add and drop partitions using the `ALTER TABLE <table name> ADD PARTITION (<partition specification>)` or `ALTER TABLE <table name> DROP PARTITION <list of partitions>` statement.

> For `LIST and `RANGE` partitioned tables, `REORGANIZE PARTITION` is not yet supported.

Reviewer: ```suggestion
For `LIST` and `RANGE` partitioned tables, `REORGANIZE PARTITION` is not yet supported.
```

`REORGANIZE PARTITION` is not yet supported.

> **Warning:**
>

> > - `EXCHANGE PARTITION` is an experimental feature. It is not recommended to use it in a production environment. To enable it, set the `tidb_enable_exchange_partition` system variable to `ON`.

Reviewer: + Warning notes are used to provide additional information to the content above. There seems no mention of `EXCHANGE PARTITION` before this warning note.
+ The warning note only serves as a reminder or alert, and should be concise enough. One or two sentences should be enough. Detailed explanation or even example are not recommended. Could you consider putting some of the information out of the warning note?

> - As an example: `ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table`, swaps the table `non_partitioned_table` with the partition `p1` in the `partitioned_table` table.
> 
> Please be sure that all rows you are exchanging into the partition are matching the partition definition, or else they will not be found and there may be other issues.
> 
> > **Warning:**
> >
> > - `EXCHANGE PARTITION` is an experimental feature. It is not yet recommended to use it in a production environment. To enable it, set the `tidb_enable_exchange_partition` system variable to `ON`.

Reviewer: ```suggestion
For example, `ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table` swaps the `non_partitioned_table` table in the `p1` partition with the `partitioned_table` table.

Ensure that all rows that you are exchanging into the partition match the partition definition; otherwise, these rows will not be found and cause unexpected issues.

> **Warning:**
>
> `EXCHANGE PARTITION` is an experimental feature. It is not recommended to use it in a production environment. To enable it, set the `tidb_enable_exchange_partition` system variable to `ON`.
```
